### ISSUE TITLE
Data module removing 0's in fips codes

### ISSUE THREAD

**Issue by kai2468** [highcharts/highcharts#16920]
#### Expected behaviour
The data module does not remove the first 0 in the fips code and the map renders correctly

#### Actual behaviour
The data module removes the first 0 in the fips code and the map does not render any of its points.

An example of a fips code in the Alabama map: 01007

The data module seems to be removing the first 0 in the codes, and so Highmaps cant find the right code to link to.

If you log mapKey and mapMap in the line I've linked below, you'll see that the codes do not match. I assume the data module is doing some custom parsing before it gets to this point and the 0 is getting removed because of it.

https://github.com/highcharts/highcharts/blob/master/ts/Series/Map/MapSeries.ts#L1057

#### Live demo with steps to reproduce
https://jsfiddle.net/vf2yeL5g/

#### Product version
Latest

**Comment by KacperMadej** [comment]
Hi @kai2468 

Thank you for reporting the problem.

_Internal note / more details:_
Data module has the source of the problem - it is not a Maps specific issue.
A solution would be to have an option to indicate the type of each column in CSV. In code there is a trace of such option - `columnTypes` is the name. It is not documented and should work when checking the raw code. However, when debugging there is a part of guessed parsing of CSV before that option is even checked.

https://github.com/highcharts/highcharts/blob/8c4f1a219435ce85cad1cbb28da55101101f66ff/ts/Extensions/Data.ts#L1150-L1152

So basically, if it is parsable it will be parsed, so there is no API option to prevent this.

A fix for the bug could be to add `columnTypes` as a API option ([we already have some code in place, a test, and a demo using it](https://github.com/highcharts/highcharts/search?q=columnTypes)).

PoC (with modified code marked with `-------` comments): https://jsfiddle.net/BlackLabel/Ljxhgve4/

**Comment by kai2468** [comment]
Hi, any news on this one?
